### PR TITLE
[stable/airflow] Only set Celery & Redis env variables when needed

### PR DESCRIPTION
**Pull request by omonnier**
The environment variables for Redis and Celery prevent
the helm chart to work in LocalExecutori mode.

They are now only exported if:
- Redis is enabled
- and Airflow executor is "Celery"

Signed-off-by: Olivier Monnier <5500018+omonnier@users.noreply.github.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/airflow/templates/configmap-env.yaml [hunk 1]
```diff
@@ -14,28 +14,36 @@ data:
...
   AIRFLOW__CELERY__WORKER_CONCURRENCY: "{{ .Values.workers.celery.instances }}"
+  {{- end }}
   AIRFLOW__CORE__DAGS_FOLDER: "{{ .Values.dags.path }}"
   AIRFLOW__CORE__BASE_LOG_FOLDER: "{{ .Values.logs.path }}"
   AIRFLOW__CORE__DAG_PROCESSOR_MANAGER_LOG_LOCATION: "{{ printf "%s/%s" .Values.logs.path "dag_processor_manager/dag_processor_manager.log" }}"
   AIRFLOW__SCHEDULER__CHILD_PROCESS_LOG_DIRECTORY: "{{ printf "%s/%s" .Values.logs.path "scheduler" }}"
   AIRFLOW__WEBSERVER__BASE_URL: "{{ .Values.ingress.web.host }}{{ .Values.ingress.web.path }}"
   # Disabling XCom pickling for forward compatibility
   AIRFLOW__CORE__ENABLE_XCOM_PICKLING: "false"
+  {{- if eq .Values.airflow.executor "Celery" }}
```
Comment (gsemet) [inline stable/airflow/templates/configmap-env.yaml]: can you move into a single block with this condition?

File: stable/airflow/templates/configmap-env.yaml [hunk 2]
```diff
@@ -14,28 +14,36 @@ data:
   POSTGRES_HOST: "{{ template "airflow.postgresql.fullname" . }}"
   POSTGRES_PORT: "{{ .Values.postgresql.service.port }}"
   POSTGRES_DB: "{{ .Values.postgresql.postgresDatabase }}"
+  {{- if .Values.redis.enabled }}
```
Comment (gsemet) [inline stable/airflow/templates/configmap-env.yaml]: 'redis.enabled=true' actually creates a redis instance, while user can still provide the url to their own one with redis.enabled=false.

You can move under the `if eq .Values.airflow.executor "Celery"` unbrella

**Review by gsemet — CHANGES_REQUESTED** [review]

File: stable/airflow/templates/configmap-env.yaml (hunk 1 above)
Comment (omonnier) [reply stable/airflow/templates/configmap-env.yaml]: done

File: stable/airflow/templates/configmap-env.yaml (hunk 2 above)
Comment (omonnier) [reply stable/airflow/templates/configmap-env.yaml]: done

**Pull request merged**